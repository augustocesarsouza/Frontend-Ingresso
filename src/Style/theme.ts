export const theme = {
  colors: {
    primaryColor: '#333', // meio-escuro
    secondaryColor: '#2634a2', // azul
    white: '#FFFFFF',
    mediumGray: '#DDDDDD',
  },
  font: {
    family: {
      default: "'Roboto'",
      secondary: "'Arial'",
    },
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};

// export interface Theme {
//   colors: {
//     primaryColor: string;
//     secondaryColor: string;
//     white: string;
//     mediumGray: string;
//   },
//   font: {
//     family: {
//       default: string;
//       secondary: string;
//     },
//     sizes: {
//       xsmall: string;
//       small: string;
//       medium: string;
//       large: string;
//       xlarge: string;
//       xxlarge: string;
//       huge: string;
//       xhuge: string;
//     },
//   },
//   media: {
//     lteMedium: string;
//   },
//   spacings: {
//     xsmall: string;
//     small: string;
//     medium: string;
//     large: string;
//     xlarge: string;
//     xxlarge: string;
//     huge: string;
//     xhuge: string;
//   },
// }
