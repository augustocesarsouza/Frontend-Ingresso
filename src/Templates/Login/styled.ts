import styled, { css } from 'styled-components';
import { Theme } from '../../Style/theme';

export const ContainerMain = styled.div`
  ${({theme}: { theme: Theme }) => css`
    font-size: ${theme.font.sizes.small};
  `}
`;