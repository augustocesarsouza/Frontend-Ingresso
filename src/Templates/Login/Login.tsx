import * as Styled from './styled';

const Login = () => {
  return (
    <Styled.ContainerMain>
      <h1>Tetse</h1>
    </Styled.ContainerMain>
  );
};

export default Login;
