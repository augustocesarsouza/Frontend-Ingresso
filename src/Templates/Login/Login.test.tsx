import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Login from './Login';
import { RenderTheme } from '../../Style/RenderTheme';

describe('Login', () => {
  test('should render Login', () => {
    const { container } = RenderTheme(<Login />);
    const h1 = screen.getByRole('heading');
    expect(h1).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
