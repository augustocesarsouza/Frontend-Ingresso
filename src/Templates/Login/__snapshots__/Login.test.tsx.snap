// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render Login 1`] = `
.c0 {
  font-size: 1.6rem;
}

<div>
  <div
    class="c0"
  >
    <h1>
      Tetse
    </h1>
  </div>
</div>
`;
